core/shared/src/main/scala/zio/logging/LogContext.scala

```diff
@@ -32,6 +32,12 @@ final case class LogContext private (private val map: Map[LogAnnotation[_], Any]
   def get[A](annotation: LogAnnotation[A]): A =
     map.get(annotation).fold(annotation.initialValue)(_.asInstanceOf[A])
 
+  /**
+   * Retrieves all annotations from the context.
+   */
+  def getAll: Map[LogAnnotation[_], Any] =
```

Comment:
returning something with `Any` make me feel sick :) 

@jdegoes what do you think?